versions/scylla/3.21.0/patch

```diff
@@ -0,0 +1,704 @@
+diff --git a/tests/integration/__init__.py b/tests/integration/__init__.py
+index d6f26acb..fb2b31ed 100644
+--- a/tests/integration/__init__.py
++++ b/tests/integration/__init__.py
+@@ -36,6 +36,7 @@ from itertools import groupby
+ import six
+ import shutil
+ 
++
+ from cassandra import OperationTimedOut, ReadTimeout, ReadFailure, WriteTimeout, WriteFailure, AlreadyExists,\
+     InvalidRequest
+ from cassandra.protocol import ConfigurationException
+@@ -44,6 +45,7 @@ from cassandra import ProtocolVersion
+ try:
+     from ccmlib.dse_cluster import DseCluster
+     from ccmlib.cluster import Cluster as CCMCluster
++    from ccmlib.scylla_cluster import ScyllaCluster as CCMScyllaCluster
+     from ccmlib.cluster_factory import ClusterFactory as CCMClusterFactory
+     from ccmlib import common
+ except ImportError as e:
+@@ -161,16 +163,21 @@ KEEP_TEST_CLUSTER = bool(os.getenv('KEEP_TEST_CLUSTER', False))
+ SIMULACRON_JAR = os.getenv('SIMULACRON_JAR', None)
+ CLOUD_PROXY_PATH = os.getenv('CLOUD_PROXY_PATH', None)
+ 
+-# Supported Clusters: Cassandra, DDAC, DSE
++# Supported Clusters: Cassandra, DDAC, DSE, Scylla
+ DSE_VERSION = None
++SCYLLA_VERSION = os.getenv('SCYLLA_VERSION', None)
+ if os.getenv('DSE_VERSION', None):  # we are testing against DSE
+     DSE_VERSION = Version(os.getenv('DSE_VERSION', None))
+     DSE_CRED = os.getenv('DSE_CREDS', None)
+     CASSANDRA_VERSION = _get_cass_version_from_dse(DSE_VERSION.base_version)
+     CCM_VERSION = DSE_VERSION.base_version
+ else:  # we are testing against Cassandra or DDAC
+-    cv_string = os.getenv('CASSANDRA_VERSION', None)
+-    mcv_string = os.getenv('MAPPED_CASSANDRA_VERSION', None)
++    if SCYLLA_VERSION:
++        cv_string = SCYLLA_VERSION
++        mcv_string = os.getenv('MAPPED_SCYLLA_VERSION', None)
```

Comment:
@dkropachev why did we need to  add  it?
What does it do?
AND where the HELL do we set it?

I'm asking because I see that the Jenkins jobs are running with this value set to 999.99.0 and the tests setup fails with:
```
Error Message
expected string or bytes-like object
-------------------- >> begin captured logging << --------------------
tests.integration: INFO: Using Cassandra version: 999.99.0
tests.integration: WARNING: FileNotFoundError: [Errno 2] No such file or directory: '/home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/ccm/test_cluster/cluster.conf' Backtrace: [<FrameSummary file /home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/__init__.py, line 492 in use_cluster>, <FrameSummary file /home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/scylla-ccm/ccmlib/cluster_factory.py, line 20 in load>]
tests.integration: DEBUG: Creating new CCM cluster, test_cluster, with args {'version': '999.99.0'}
ccm: DEBUG: Started cluster 'test_cluster' version 4.7.dev installed in /home/jenkins/.ccm/scylla-repository/999.99.0
--------------------- >> end captured logging << ---------------------
Stacktrace
  File "/home/jenkins/.local/lib/python3.7/site-packages/nose/suite.py", line 210, in run
    self.setUp()
  File "/home/jenkins/.local/lib/python3.7/site-packages/nose/suite.py", line 293, in setUp
    self.setupContext(ancestor)
  File "/home/jenkins/.local/lib/python3.7/site-packages/nose/suite.py", line 316, in setupContext
    try_run(context, names)
  File "/home/jenkins/.local/lib/python3.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/standard/test_authentication.py", line 39, in setup_module
    use_singledc(start=False)
  File "/home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/__init__.py", line 403, in use_singledc
    use_cluster(CLUSTER_NAME, [3], start=start, workloads=workloads)
  File "/home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/__init__.py", line 540, in use_cluster
    if Version(cassandra_version) >= Version('2.2'):
  File "/home/jenkins/.local/lib/python3.7/site-packages/packaging/version.py", line 264, in __init__
    match = self._regex.search(version)
expected string or bytes-like object
-------------------- >> begin captured logging << --------------------
tests.integration: INFO: Using Cassandra version: 999.99.0
tests.integration: WARNING: FileNotFoundError: [Errno 2] No such file or directory: '/home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/ccm/test_cluster/cluster.conf' Backtrace: [<FrameSummary file /home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/python-driver-datastax/tests/integration/__init__.py, line 492 in use_cluster>, <FrameSummary file /home/jenkins/slave/workspace/scylla-staging/Oren/python-driver-matrix-test@2/scylla-ccm/ccmlib/cluster_factory.py, line 20 in load>]
tests.integration: DEBUG: Creating new CCM cluster, test_cluster, with args {'version': '999.99.0'}
ccm: DEBUG: Started cluster 'test_cluster' version 4.7.dev installed in /home/jenkins/.ccm/scylla-repository/999.99.0
--------------------- >> end captured logging << ---------------------
```

Reply:
We use `SCYLLA_VERSION` as means of providing info where to get scylla package, it could de something like `'unstable/master:latest'`.
In such case you can get version of scylla that this package holds, before you install it.

While test infrastructure needs to have exact `scylla version` to skip/not to skip tests before cluster is deployed.
In such a case we get `scylla version` from `MAPPED_SCYLLA_VERSION`.

Backtrace you provided is quite strange, since I see that `MAPPED_SCYLLA_VERSION`, is actually provided, but at the same time `cassandra_version` is None.
Probably our patches were applied improperly and that is why it had failed.

Reply:
But what is 999.99.0?
We do we set it like that?

Reply:
@roydahan , we set it like this to pass all possible checkers that could disable features because of the version

Reply:
ok, thanks.